src/gtk/window.cpp

```diff
@@ -4240,6 +4243,14 @@ bool wxWindowGTK::GTKShowFromOnIdle()
     return false;
 }
 
+#ifdef __WINDOWS__
+WXHWND wxWindowGTK::GetHWND() const {
+    // There isn't an underlying HWND if the widget hasn't been realized yet.
+    gtk_widget_realize(m_widget);
```

Comment:
This is more serious, IMO it's really unexpected for an accessor like this to have significant side effects.

Would it be a problem to just call this function only if the window had been already realized?

Reply:
Would need to add an extra check on the first gtk function, since it'll return null if not (and passing that to gdk will cause *bad things*), but that's easy enough.

However, that'd require a way (for my use case at least) to be able to realize the GTK Window through wxWidgets. I'll see if there's a function for that already. If not, I'd want to add one.

Reply:
I don't see a method to "force" realize a gtk window.

Are you aware if there's something like this that exists and/or what sort of side effects would need to be considered by creating another function (GTKRealizeWidget() for example) to do this?